core/src/test/java/org/elasticsearch/action/support/master/IndexingMasterFailoverIT.java

```diff
@@ -73,9 +73,10 @@ public void testMasterFailoverDuringIndexingWithMappingChanges() throws Throwabl
                 .put("discovery.zen.join_timeout", "10s")  // still long to induce failures but to long so test won't time out
                 .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "1s") // <-- for hitting simulated network failures quickly
                 .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 2)
+                .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0")  // <-- we wait for cluster formation at the end
```

Comment:
> My goal here is to remove the complexity and asynchronicity of node starting. Adding that back will defeat what I wanted to achieve.

I'm not arguing that the startNode(s) APIs should expose asynchronicity. I'm talking about how the APIs should internally work. There I think they should behave in the way how you would normally start up nodes in real-life. An API call startNodes(3) should start 3 nodes and, after the method returns, have all 3 nodes in a state where they're started (their Node#start() methods returned). Internally, the API should follow what I would do in a real-world scenario: Start the 3 nodes and wait for all 3 to be ready. I wouldn't go and use the INITIAL_STATE_TIMEOUT_SETTING to achieve that but just boot them up in parallel.

> To be clear - if you use that method when min_master_nodes is auto managed you don't need to do anything. It's only if you "own" that setting than you have to also deal with the sequential nature of starting (unless you use this setting which IMO a valid use case and the reason for it's existence).

It does not matter whether `min_master_nodes` is auto-managed or not. In both those cases, there is no need for `INITIAL_STATE_TIMEOUT_SETTING` if it's implemented in the way I outlined.

> the reason for it's existence

No. It's used as an internal setting to start up tribe nodes. It's not even mentioned in our documentation.

I think that this is the wrong approach and that there is a cleaner way to do this.